pkg/odo/cli/project/list.go

```diff
@@ -0,0 +1,87 @@
+package project
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/project"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+)
+
+const listRecommendedCommandName = "list"
+
+var (
+	listExample = ktemplates.Examples(`
+	# List all the projects
+    %[1]s`)
+	listLongDesc = ktemplates.LongDesc(`
+	List all the projects
+`)
+)
+
+// ProjectListOptions encapsulates the options for the odo project list command
+type ProjectListOptions struct {
+	*genericclioptions.Context
+}
+
+// NewProjectListOptions creates a new ProjectListOptions instance
+func NewProjectListOptions() *ProjectListOptions {
+	return &ProjectListOptions{}
+}
+
+// Complete completes ProjectListOptions after they've been created
+func (plo *ProjectListOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	plo.Context = genericclioptions.NewContext(cmd)
+	return
+}
+
+// Validate validates the ProjectListOptions based on completed values
+func (plo *ProjectListOptions) Validate() (err error) {
+	return
+}
+
+// Run contains the logic for the odo project list command
+func (plo *ProjectListOptions) Run() (err error) {
+
+	projects, err := project.List(plo.Client)
+	if err != nil {
+		return fmt.Errorf("Project catalog is not enabled within your cluster: %v", err)
```

Comment:
Where is this coming from? There is no such thing as `Project catalog`. Look at the command pre-refactoring.

Reply:
ack, resolved